projects/spectator/src/lib/spectator/spectator.ts

```diff
@@ -45,4 +45,79 @@ export class Spectator<C> extends DomSpectator<C> {
     // Force cd on the host component for cases such as: https://github.com/ngneat/spectator/issues/539
     this.detectChanges();
   }
+
+  public get deferBlocks(): DeferBlocks {
```

Comment:
Any reason we need to create a new reference object (i.e the return value of _deferBlocksForGivenFixture) each time we use this getter?

Reply:
I am not sure if I fully understand you, so I'll do my best answering :) 

My first approach was to use the `deferBlock` both from the parent level and the children ones.
I didn't like, though, the idea of exposing an argument for the sake of internal usage.

```ts
// Parent level
public deferBlock(deferBlockIndex = 0, deferBlockFixtures = this.fixture.getDeferBlocks()): DeferBlocks {
```

// Children
![image](https://github.com/ngneat/spectator/assets/7045092/81a582d8-3659-4758-ae4f-3c0aa6fff45d)

To avoid that, I created a private function.

I feel that you were referring to something else though.

Reply:
It doesn't matter anymore because we are replacing this API to [this](https://github.com/ngneat/spectator/pull/641/files#r1510292191).